master fails to build on macOS

MacOS 11.6.2, Xcode-13.2.1.

Tried with Cabal-3.6.2.0 (GHC-9.0.1 and 9.0.2), and Stack-2.7.3 (GHC-8.10.7).

With Cabal, building tests fails:
```
$ cabal test
.  .  .
      Blake2b-512
        1:                                   OK
        2:                                   OK
        3:                                   dyld: lazy symbol binding failed: Symbol not found: __cryptonite_blake2s_init
  Referenced from: /Users/ur20980/src/cryptonite/dist-newstyle/build/x86_64-osx/ghc-9.0.1/cryptonite-0.29/build/libHScryptonite-0.29-inplace-ghc9.0.1.dylib
  Expected in: flat namespace

dyld: Symbol not found: __cryptonite_blake2s_init
  Referenced from: /Users/ur20980/src/cryptonite/dist-newstyle/build/x86_64-osx/ghc-9.0.1/cryptonite-0.29/build/libHScryptonite-0.29-inplace-ghc9.0.1.dylib
  Expected in: flat namespace


Test suite test-cryptonite: FAIL
Test suite logged to:
/Users/ur20980/src/cryptonite/dist-newstyle/build/x86_64-osx/ghc-9.0.1/cryptonite-0.29/t/test-cryptonite/test/cryptonite-0.29-test-cryptonite.log
0 of 1 test suites (0 of 1 test cases) passed.
cabal: Tests failed for test:test-cryptonite from cryptonite-0.29.
```

```
$ nm /Users/ur20980/src/cryptonite/dist-newstyle/build/x86_64-osx/ghc-9.0.1/cryptonite-0.29/build/libHScryptonite-0.29-inplace-ghc9.0.1.dylib | grep __cryptonite_blake2s_init
U __cryptonite_blake2s_init
U __cryptonite_blake2s_init_param
```

WIth Stack, linking tests fails:
```
Linking .stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/test-cryptonite/test-cryptonite ...
Undefined symbols for architecture x86_64:
  "__cryptonite_blake2s_final", referenced from:
      _cryptonite_blake2s_finalize in libHScryptonite-0.29-5gTqpPHk8HTCvUfrUbEit5.a(cryptonite_blake2s.o)
  "__cryptonite_blake2s_init", referenced from:
      _cryptonite_blake2s_init in libHScryptonite-0.29-5gTqpPHk8HTCvUfrUbEit5.a(cryptonite_blake2s.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`clang' failed in phase `Linker'. (Exit code: 1)
Progress 18/19            

--  While building package cryptonite-0.29 (scroll up to its section to see the error) using:
      /Users/ur20980/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_3.2.1.0_ghc-8.10.7 --builddir=.stack-work/dist/x86_64-osx/Cabal-3.2.1.0 build lib:cryptonite test:test-cryptonite --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

Would complete test log help? I'll post it if so.

--- a/cbits/blake2/sse/blake2s.c
+++ b/cbits/blake2/sse/blake2s.c
@@ -72,7 +72,7 @@ static void blake2s_increment_counter( blake2s_state *S, const uint32_t inc )
 }
 
 /* init2 xors IV with input parameter block */
-int blake2s_init_param( blake2s_state *S, const blake2s_param *P )
+int _cryptonite_blake2s_init_param( blake2s_state *S, const blake2s_param *P )
 {
   size_t i;
   /*blake2s_init0( S ); */
@@ -90,7 +90,7 @@ int blake2s_init_param( blake2s_state *S, const blake2s_param *P )
 
 
 /* Some sort of default parameter block initialization, for sequential blake2s */
-int blake2s_init( blake2s_state *S, size_t outlen )
+int _cryptonite_blake2s_init( blake2s_state *S, size_t outlen )
 {
   blake2s_param P[1];
 
@@ -110,11 +110,11 @@ int blake2s_init( blake2s_state *S, size_t outlen )
   memset( P->salt,     0, sizeof( P->salt ) );
   memset( P->personal, 0, sizeof( P->personal ) );
 
-  return blake2s_init_param( S, P );
+  return _cryptonite_blake2s_init_param( S, P );
 }
 
 
-int blake2s_init_key( blake2s_state *S, size_t outlen, const void *key, size_t keylen )
+int _cryptonite_blake2s_init_key( blake2s_state *S, size_t outlen, const void *key, size_t keylen )
 {
   blake2s_param P[1];
 
@@ -136,7 +136,7 @@ int blake2s_init_key( blake2s_state *S, size_t outlen, const void *key, size_t k
   memset( P->salt,     0, sizeof( P->salt ) );
   memset( P->personal, 0, sizeof( P->personal ) );
 
-  if( blake2s_init_param( S, P ) < 0 )
+  if( _cryptonite_blake2s_init_param( S, P ) < 0 )
     return -1;
 
   {
@@ -233,7 +233,7 @@ int _cryptonite_blake2s_update( blake2s_state *S, const void *pin, size_t inlen 
   return 0;
 }
 
-int blake2s_final( blake2s_state *S, void *out, size_t outlen )
+int _cryptonite_blake2s_final( blake2s_state *S, void *out, size_t outlen )
 {
   uint8_t buffer[BLAKE2S_OUTBYTES] = {0};
   size_t i;
@@ -275,15 +275,15 @@ int blake2s( void *out, size_t outlen, const void *in, size_t inlen, const void 
 
   if( keylen > 0 )
   {
-    if( blake2s_init_key( S, outlen, key, keylen ) < 0 ) return -1;
+    if( _cryptonite_blake2s_init_key( S, outlen, key, keylen ) < 0 ) return -1;
   }
   else
   {
-    if( blake2s_init( S, outlen ) < 0 ) return -1;
+    if( _cryptonite_blake2s_init( S, outlen ) < 0 ) return -1;
   }
 
   _cryptonite_blake2s_update( S, ( const uint8_t * )in, inlen );
-  blake2s_final( S, out, outlen );
+  _cryptonite_blake2s_final( S, out, outlen );
   return 0;
 }
 
@@ -330,7 +330,7 @@ int main( void )
       size_t mlen = i;
       int err = 0;
 
-      if( (err = blake2s_init_key(&S, BLAKE2S_OUTBYTES, key, BLAKE2S_KEYBYTES)) < 0 ) {
+      if( (err = _cryptonite_blake2s_init_key(&S, BLAKE2S_OUTBYTES, key, BLAKE2S_KEYBYTES)) < 0 ) {
         goto fail;
       }
 
@@ -344,7 +344,7 @@ int main( void )
       if ( (err = _cryptonite_blake2s_update(&S, p, mlen)) < 0) {
         goto fail;
       }
-      if ( (err = blake2s_final(&S, hash, BLAKE2S_OUTBYTES)) < 0) {
+      if ( (err = _cryptonite_blake2s_final(&S, hash, BLAKE2S_OUTBYTES)) < 0) {
         goto fail;
       }
 

